src/model.rs

```diff
@@ -93,7 +137,7 @@ impl<B: Backend<FloatElem = f32>> Model<B> {
             );
             let s_forget = self.stability_after_failure(stability, new_d.clone(), r);
             let new_s = s_recall.mask_where(rating.equal_elem(1), s_forget);
-            (new_s.clamp(0.1, 36500.0), new_d)
+            (new_s.clamp(0.1, 36500.0), weight_clipper(new_d))
```

Comment:
It's not the position to call this function. Weight clipper should be applied to the parameters.

Reply:
Could you call weight_clipper() in the end of forward()?

Reply:
Like that?